cubi_tk/snappy/common.py

```diff
@@ -37,6 +23,124 @@ class CouldNotFindBioMedSheet(Exception):
+
+
+def get_workflow_snakefile_object_name(snakefile_path):
+    """Find the Workflow implementation object name.
+
+    :param snakefile_path: Path to snakefile for workflow.
+    :type snakefile_path: str, pathlib.Path
+
+    :return: str Name of the implementation class or None if nothing as been found.
+    """
+
+    with snakefile_path.open() as f:
+        if m := re.search(r"wf\s*=\s*(\w+)\(", f.read()):
+            module_name = m.group(1)
+            return module_name
+    return None
+
+
+def load_workflow_step_configuration(workflow_step_path):
+    """Load snappy configuration and resolve any refs.
+
+    :param workflow_step_path: Path to snappy config yaml.
+    :type workflow_step_path: str, pathlib.Path
+
+    :return: Tuple of config, lookup paths and configuration paths. If no config is found, a tuple of None is returned.
+    """
+
+    config_path = pathlib.Path(workflow_step_path) / "config.yaml"
+
+    if not config_path.exists():
+        return (None, None, None)
+
+    with open(str(config_path)) as stream:
+        config_data = yaml.safe_load(stream)
+
+    config, lookup_paths, config_paths = expand_ref(
+        str(config_path),
+        config_data,
+        [str(workflow_step_path), str(workflow_step_path.parent / ".snappy_pipeline")],
+    )
+    return config, lookup_paths, config_paths
+
+
+def get_snappy_config(snappy_root_dir):
+    """Get snappy configuration.
+
+    :param snappy_root_dir: Path to snappy root directory.
+    :type snappy_root_dir: str, pathlib.Path
+
+    :return: Returns loaded snappy configuration.
+    """
+    snappy_config = snappy_root_dir / ".snappy_pipeline" / "config.yaml"
+    with open(snappy_config, "r") as stream:
+        config = yaml.safe_load(stream)
+    return config
+
+
+def get_workflow_step_dependencies(workflow_step_path):
+    """Find workflow dependencies for the given workflow step.
+    :param workflow_step_path: Path to the workflow step.
+    :type workflow_step_path: str, pathlib.Path
+
+    :return: List of dependencies for the given workflow step.
+    """
+    workflow_step_path = pathlib.Path(workflow_step_path)
+
+    config, lookup_paths, config_paths = load_workflow_step_configuration(workflow_step_path)
+    if config is None:
+        raise RuntimeError(f"Could not load workflow step confiuration for {workflow_step_path}")
+
+    step_name = config["pipeline_step"]["name"]
+
+    step_module_paths = get_available_snappy_workflow_paths()
+    module_path = step_module_paths[step_name]
+    module_config_path = module_path / "Snakefile"
+    # get the name of the workflow step class name
+    obj_name = get_workflow_snakefile_object_name(module_config_path)
+    if obj_name is None:
+        raise RuntimeError(f"Could not find workflow object for {step_name}")
+
+    workflow_module = importlib.import_module(f".{step_name}", "snappy_pipeline.workflows")
+    workflow_class = getattr(workflow_module, obj_name)
```

Comment:
I'm not really happy with having to parse the snappy source code to get the name of the workflow/Step-Manager object here.

From what I can see these objects are almost always the Camel-Case (+Workflow) of the step_name. Maybe we can rely on that? Neither is really a neat solution though.
Either way we should assure that the `name` attribute of the imported class matches the desired step_name.